src/reporter.ts

```diff
@@ -127,4 +148,24 @@ export class Reporter {
       });
     });
   }
+
+  setLaunchStatus(status: string): void {
+    this.customLaunchStatus = status;
+  }
+
+  getCurrentSuiteId(): string {
+    return this.suiteIds.length ? this.suiteIds[this.suiteIds.length - 1] : undefined;
+  }
+
+  getCurrentTestItemId(): string {
+    return (
+      (this.testItems.length && this.testItems[this.testItems.length - 1].id) ||
+      this.getCurrentSuiteId()
+    );
+  }
+
+  setStatus({ status }: ObjUniversal): void {
+    const testItemId = this.getCurrentTestItemId();
+    this.testData[testItemId] = status;
```

Comment:
Seems that `this.testData` contains info about the suite (according to `reportFixtureStart`), so it may hold suite status.
For steps I suggest to use `testItems` array instead (to simply hold other additional data like attributes, testCaseId, etc.)
You can extract testItem object from `testItems` and set status for it, if it is not exists just use `testData` for suite.